### DISCUSSION TITLE
Update mermaid to latest version

### DISCUSSION THREAD

**Discussion by Zerotask** [community/community#70672] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Feature Area

Issues

### Body

I noticed that GitHub is not using the latest version of [mermaidjs](https://github.com/mermaid-js/mermaid) since some things I were using are not displayed properly.
Therefore it would be nice to update it to the latest version.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @Zerotask ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Projects and Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!

**Reply by Zerotask** [reply]
Hello @queenofcorgis 

Thank you for your message. I posted it on the Issues category, since the problem occurs in an issue. Issues render mermaid syntax, but the latest charts/diagrams are missing.

**Comment by baynezy** [comment]
What is the GitHub roadmap for upgrading things like mermaid? Is that published anywhere? It makes it hard to commit to using features if they are going to languish and go stale.

**Comment by baynezy** [comment]
Currently the installed version mermaid is:
```mermaid
info
```

The above is dynamic, but the version is `10.0.2` at the time of writing, which was released on 2nd March 2023. The current version is `10.6.1` released on 6th November 2023

**Reply by praseodym** [reply]
v10.8.0 as of today 🥳

**Comment by thibaudcolas** [comment]
In addition to upgrading, it’d be nice to add which version is currently in use in the documentation for GitHub. See #37498 for a list of reasons why this would be helpful. It appears that #86514 for example is just a matter of the GitHub-bundled version being out-of-date.

**Comment by quilicicf** [comment]
Any news on this? 
We keep getting bitten by the graphs-that-work-on-my-machine-but-not-on-GitHub issue :-(

**Comment by fdwr** [comment]
Hmm, I'm also seeing v 10.0.2, but my existing mermaid diagram just broke some time the past week or few since I last viewed the diagram on GitHub. Hypertext does not display correctly, and instead of `Test["<b>Test</b>"]` displaying as **Test**, I now literally see `<b>Test</b>` in the diagram. 😢

**Comment by baynezy** [comment]
This is my biggest problem with mermaid. It is a great idea, but spotty support makes it a lottery to work with. I have beautiful diagrams that I craft in my IDE or on mermaid.live and then they end up having to be converted to a PNG for GitHub and DevOps wiki as the support isn't there. Which completely negates the point.

**Reply by fdwr** [reply]
What tool do you use for conversion? Are SVG or HTML valid outputs?

**Reply by quilicicf** [reply]
Probably [mermaid-cli](https://www.npmjs.com/package/@mermaid-js/mermaid-cli) ?
It does support SVG.

**Reply by fdwr** [reply]
> Probably [mermaid-cli](https://www.npmjs.com/package/@mermaid-js/mermaid-cli)

@quilicicf Arigatou - will try it.

**Reply by baynezy** [reply]
Sorry I use the Mermaid website. mermaid.live. It has an export function.

**Comment by ryanobjc** [comment]
I am using sequenceDiagrams in mermaid and looks like the version github has embedded doesn't support the 'create participant X' feature, which is a real bummer!

**Reply by moop-moop** [reply]
I totally agree with you. And I even wanted to use sequence diagrams to better document our GitHub Actions and Workflows.
That feature started in Mermaid 10.3

**Comment by abernier** [comment]
An amazing feature 
![image](https://github.com/community/community/assets/76580/1df8e117-6d0c-4e61-9de3-abb287a7417d)
https://github.com/mermaid-js/mermaid/pull/2885 has been merged/is coming to mermaid, hope gh will bump the version so everyone can benefit from soon

**Comment by praseodym** [comment]
Looks like GitHub is on Mermaid v10.8.0 now!

**Reply by abernier** [reply]
does not seem to work in 10.8.0 (which GitHub is indeed running on): https://github.com/mermaid-js/mermaid/pull/2885#issuecomment-1962889769

**Reply by praseodym** [reply]
That's because there hasn't been any Mermaid releases since the Katex feature has been merged. The Katex feature has been merged on 13 February 2024, and the current latest release of Mermaid was on 2 February 2024. 

Once a new Mermaid release has been made, we'll have to wait for GitHub to update its Mermaid version. Hopefully that won't be too long!

**Reply by quilicicf** [reply]
Damn, you are right!
```mermaid
info
```

**Comment by mhartzell-incomm** [comment]
Mermaid 11.3 is out and DANG I need me some new nodes.

**Comment by notesbytom** [comment]
v11.4.1 (deployed on GitHub as of 2025-04-08) has a bug where Open Links/Edges `A---B` display with an Arrow Head like `A-->B`. I believe this is fixed in v11.5.0 https://github.com/mermaid-js/mermaid/pull/6093#issuecomment-2734046994 . Hopefully a fixed version will be deployed soon.

**Comment by diegosasw** [comment]
[11.15.0](https://github.com/mermaid-js/mermaid/releases/tag/mermaid%4011.15.0)
Current version is now 11.15.0, it contains event modeling diagrams.
Is there a way to upgrade mermaid for a GitHub repo instead of waiting for a global upgrade which seems to take too long?

**Comment by sunanana** [comment]
Mermaid v11.17.0 added subgraph support to ER diagrams (mermaid-js/mermaid#7792), which allows grouping entities into logical domains — something that has been requested for a long time (mermaid-js/mermaid#5168, mermaid-js/mermaid#7417).

However, the Mermaid version currently bundled on GitHub doesn't support it yet. As of today, GitHub is running Mermaid v11.16.1 (checked via the `info` diagram), so it's just one version behind. The `subgraph` / `end` keywords are parsed as plain entities, so the diagram below renders incorrectly:

```mermaid
erDiagram
    subgraph auth [Auth]
        users
        sessions
    end

    users ||--o{ sessions : "logs in"
```

It would be great if the bundled Mermaid version could be updated to v11.17.0 or later so that ER diagram subgraphs render properly in Markdown files, issues, and PRs.
